src/Tasks/Exec.cs

```diff
@@ -611,12 +611,26 @@ protected internal override void AddCommandLineCommands(CommandLineBuilderExtens
 
                     // If for some crazy reason the path has a & character and a space in it
                     // then get the short path of the temp path, which should not have spaces in it
-                    // and then escape the &
                     if (batchFileForCommandLine.Contains("&") && !batchFileForCommandLine.Contains("^&"))
                     {
                         batchFileForCommandLine = NativeMethodsShared.GetShortFilePath(batchFileForCommandLine);
-                        batchFileForCommandLine = batchFileForCommandLine.Replace("&", "^&");
                     }
+
+                    StringBuilder fileName = StringBuilderCache.Acquire(batchFileForCommandLine.Length);
+
+                    // Escape any '(', ')', or '&'
+                    for (int i = 0; i < batchFileForCommandLine.Length; i++)
+                    {
+                        char c = batchFileForCommandLine[i];
+
+                        if ((c == '(' || c == ')' || c == '&') && (i == 0 || batchFileForCommandLine[i - 1] != '^'))
```

Comment:
`^` just before the character does not mean that the character is escaped. The escaped character may be `^` itself.

Example: `^^(` the open paren won't be escaped.

Reply:
Good point. @BenVillalobos, can you add something to verify that the previous character was not ^? And I think putting it behind a change wave is reasonable.

Reply:
Turns out there's some behavior I didn't expect with `cmd /C`. Check this out:

Notice that having a carat escapes an open and closing parens?
```
cmd /C "c:\hello^()\"
'c:\hello()\' is not recognized...
```

Not two open parens though.
```
cmd /C "c:\hello^(()\"
'c:\hello(' is not recognized...
```

What about some closed parens?
```
C:\Users\bevillal>cmd /C "c:\hello^)))))\"
'c:\hello)))))\' is not recognized <--- What??
```

What if we put stuff inside the parens?
```
cmd /C "c:\hello^(ab^c)\"
'c:\hello(abc)\' is not recognized
```

Okay so we keep parsing even after seeing a '^' between escaped parens. But not if those parens are escaped.
```
C:\Users\bevillal>cmd /C "c:\hello(ab^c)\"
'c:\hello' is not recognized
```

Reply:
### The plot thickens!

Some more test cases I played with during standup.

@marcpopMSFT suggested testing with `Program Files(x86)`, which was a great idea. I created c:/temp/test folder(x86) and... `cmd /C "C:\temp\test folder(x86)\a.txt"` works.

Here are some other variations:

No space in folder: `cmd /C "C:\temp\testfolder(x86)\a.txt"`: `'C:\temp\testfolder' is not recognized`
Space and x86: `cmd /C "C:\temp\test folder(x86)\a.txt"`: Works
Space and no x86: `cmd /C "C:\temp\test folder(123)\a.txt"`: Works

Reply:
Sounds like everything is escaped if there's a space—we're detecting that separately? If so, and you can find where that's happening, that might be a much easier fix: detect characters other than spaces in addition to spaces.

Reply:
> Sounds like everything is escaped if there's a space

If only 😕
`cmd /C "C:\temp - Copy\testfolder(x86)\a^.txt"` fails
`cmd /C "C:\temp - Copy\testfolder(x86)\a(.txt"` succeeds
`cmd /C "C:\temp - Copy\testfolder(x86)\a(()()$.txt"` succeeds.
It definitely breaks on unescaped `&` and `^`

Other fun fact:
Copy pasting a file called `a(()()$.txt` resulted in `a(()$ - Copy.txt` being generated.

Reply:
Notes from PR review, consider the following:
- Consider just executing the file.
- Have an optional metadata that either:
	- 	Is some string containing for specific characters to escape
	- 	Is some boolean flag that will escape all special characters except the escape character itself (or some variation of that to give a better user experience)
- Is there some existing library to PInvoke into?
- Consider what the `/S` flag does

Reply:
The purpose of `/s` is apparently to handle commands that have quotes within them. Something like `cmd /s /c ""command1.exe" > "some/path with/spaces""` (notice the quotes around the entire expression). It looks like if we passed `/s` and surrounded the entire command in an extra set of quotes, it would properly execute the batch file if it had parens, but not improperly escaped paths. So that's a no-go.

I'll implement this by allowing a custom metadata string for "characters to escape". This is the most general solution that allows customers to fix issues with characters in their paths.

Reply:
So I've implemented the custom metadata escaping, but accounting for items that are "already escaped" is something I'm willing to make an issue for and tackle that when the demand arises. There are too many niche scenarios to handle (see above) for a scenario that is much less likely to occur for customers.

Reply:
I'm still in favor of making Exec just execute the command instead of writing and reading from a batch file, but at this point, it's probably better to wait for rainersigwald to tell us why that's a no-go. Will review this version instead.